### DISCUSSION TITLE
How to remove whitespace around the = when converting from yaml to props and change the . to any desired character?

### DISCUSSION THREAD

**Discussion by saurin-tech** [mikefarah/yq#1297] [Q&A] [answered]
I'm trying to convert a yaml file to properties file.  Everything seems to work fine but I would like to make two modifications to the properties file.

1. I would like to remove the white spaces around the "=" sign.  
For example:  
I would like to have `x.y = value` turned into `x.y=value`

3. Is there away to change the "." into any desired characters.
For example:
`x.y = value` be turned into `x_y=value`

Thank you!

**Comment by andry81** [comment]
> 2\. Is there away to change the "." into any desired characters.
>         For example:
>         `x.y = value` be turned into `x_y=value`

Just curious. Why not to just change it before yq call?

**Reply by saurin-tech** [reply]
Because it is in the yaml format and I'm using yq to convert to properties format. 

`x:
  y: value`

**Comment by mikefarah** [comment, marked as answer]
The properties encoder I'm using automatically puts the space around the equals sign - and there is no way to control that. According to the properties spec however, this whitespace is ignored (https://docs.oracle.com/cd/E23095_01/Platform.93/ATGProgGuide/html/s0204propertiesfileformat01.html). 

That said, you can use an expression to generate your own format, including the '_' instead of '.':

```bash
yq '..  | select(tag != "!!map" and tag != "!!seq") | 
  ( (path | join("_")) + "=" + .)
' input.yaml
```

Explanation:
- `..  | select(tag != "!!map" and tag != "!!seq")` find all the nodes that are scalars (not maps or arrays)
- `( (path | join("_")) + "=" + .)` grab the path array of the node, join with "_", add "=" and the value "."
